proposals/0039-debugbreak.md

```diff
@@ -56,4 +56,70 @@ void main(uint GI : SV_GroupIndex) {
 
 This aligns with C/C++ conventions that our users are already familiar with.
 
+## Detailed Design
+
+This proposal introduces a new HLSL `DebugBreak` intrinsic which has a
+runtime-defined behavior to facilitate shader debugging workflows. If the
+runtime does not support or is not configured to enable support for the
+corresponding DXIL instruction must be treated as a no-op by the driver.
+
+### HLSL Surface
+
+A new `DebugBreak` function is added with the signature:
+
+```
+void DebugBreak();
+```
+
+A new header `assert.h` is added and included with the compiler packaging which
+implements the `assert` macro:
+
+```c
+#if NDEBUG
+#define assert(cond) do { } while(false)
+#else
+#define assert(cond) do { if (!cond) DebugBreak();} while(false)
+#endif
+```
+
+### DXIL Lowering
+
+This change introduces a new DXIL operation:
+
+
+``` llvm
+declare void @dx.op.debugBreak(
+  immarg i32             ; opcode
+)
+```
+
+This DXIL operation must be treated as `convergent` even though it is not to
```

Comment:
this operation can't be moved but it doesn't prevent code being moved? (see line 121)